### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by leujas** [sebp/scikit-survival#393] [General]
Hi, I was wondering if scikit-survival could be distributed under an MIT license instead of GPL as GPL has very stringent copyleft requirements that prevent others from confidently building new code without also adopting the GPL license. Thanks.

**Comment by sebp** [comment]
I understand that GPL isn't ideal for everyone, but I do want to ensure 
that any changes made to scikit-survival will remain open-source. Hence, 
I do not want to a move to a permissive license.

**Reply by leujas** [reply]
Thanks for quick response. That makes sense.  Would you consider GPL with Classpath Exception? It would still mean any changes to the scikit-survival would remain open-source, and allows other libraries that simply call it at runtime to have their own license. Thanks.

**Comment by sebp** [comment]
I'm not sure if that's really necessary. As long as you do not 
distribute code from scikit-survival, you should be fine.

**Reply by leujas** [reply]
Thanks Seb, I’m still quite concerned simply linking to the library may trigger the copyleft requirement. I’m a fan of the library so would love to use it if possible. The Classpath Exemption provides clarity on linking the library, so would be grateful if it could be added to the license. Thank you for your consideration. 

Classpath exemption: 

As a special exception, the copyright holders of this library give you permission to link this library with independent modules to produce an executable, regardless of the license terms of these independent modules, and to copy and distribute the resulting executable under terms of your choice, provided that you also meet, for each linked independent module, the terms and conditions of the license of that module. An independent module is a module which is not derived from or based on this library. If you modify this library, you may extend this exception to your version of the library, but you are not obligated to do so. If you do not wish to do so, delete this exception statement from your version.
